hub-js/src/local/mutation/create-type-instances.ts

```diff
@@ -0,0 +1,328 @@
+} from "./cypher-errors";
+import { logger } from "../../logger";
+import { builtinStorageBackendDetails } from "./register-built-in-storage";
+
+const genAdjsColorsAndAnimals: Config = {
+  dictionaries: [adjectives, colors, animals],
+  separator: "_",
+  length: 3,
+};
+
+export interface CreateTypeInstancesArgs {
+  in: CreateTypeInstancesInput;
+}
+
+interface AliasMapping {
+  [key: string]: string;
+}
+
+export type TypeInstanceInput = Omit<CreateTypeInstanceInput, "backend"> & {
+  backend: TypeInstanceBackendDetails;
+};
+
+export async function createTypeInstances(
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  _: any,
+  args: CreateTypeInstancesArgs,
+  context: Context
+) {
+  const { typeInstances, usesRelations } = args.in;
+  const typeInstancesInput = typeInstances.map(setDefaults);
+  try {
+    validate(typeInstancesInput);
+  } catch (e) {
+    const err = e as Error;
+    throw new Error(`Validation failed: ${err.message}`);
+  }
+
+  const neo4jSession = context.driver.session();
+
+  let externallyStored: DeleteInput[] = [];
+  try {
+    return await neo4jSession.writeTransaction(async (tx: Transaction) => {
+      const createAliasMappingsResult = await createTypeInstancesInDB(
+        tx,
+        typeInstancesInput
+      );
+
+      const storeInput = getExternallyStoredValues(
+        createAliasMappingsResult,
+        typeInstancesInput
+      );
+      externallyStored = storeInput;
+      const updatedContexts = await context.delegatedStorage.Store(
+        ...storeInput
+      );
+
+      await updateTypeInstancesContextInDB(tx, updatedContexts);
+
+      await setTypeInstanceRelationsInDB(
+        tx,
+        createAliasMappingsResult,
+        usesRelations
+      );
+
+      return Object.entries(createAliasMappingsResult).map((entry) => ({
+        alias: entry[0],
+        id: entry[1],
+      }));
+    });
+  } catch (e) {
+    // Ensure that data is deleted in case of not committed transaction
+    await context.delegatedStorage.Delete(...externallyStored);
+
+    const err = e as Error;
+    throw new Error(`failed to create the TypeInstances: ${err.message}`);
+  } finally {
+    await neo4jSession.close();
+  }
+}
+
+function validate(input: TypeInstanceInput[]) {
+  const aliases = input
+    .filter((x) => x.alias !== undefined)
+    .map((x) => x.alias);
+  if (new Set(aliases).size !== aliases.length) {
+    throw new Error(
+      "Duplicated TypeInstance aliases. Please ensure that each TypeInstance alias is unique."
+    );
+  }
+
+  if (input.length !== aliases.length) {
+    throw new Error(
+      "Missing TypeInstance aliases. Please ensure that each TypeInstance has unique alias."
+    );
+  }
+
+  const notAllowedBackendCtx = input.filter(
+    (x) => x.backend.id === BUILTIN_STORAGE_BACKEND_ID && x.backend.context
+  );
+  if (notAllowedBackendCtx.length) {
+    throw new Error("Built-in storage backend does not accept context");
```

Comment:
It would be great to add TI alias or Backend ID to ensure user knows which exactly backend doesn't support context.

With such query:

```graphql
mutation CreateTypeInstances {
  createTypeInstances(
    in: {
     typeInstances: [
      {
        alias: "helm-release"
        typeRef: { path: "cap.type.helm.chart.release", revision: "0.1.0" }
        value: {
          key: "test" # same as it was
        }
        backend: {
          id: "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa" # it's like that already.
          context: { # new property of type `Any!`
            name: "release-name",
            namespace: "release-namespace",
          }
        }
      }  
      {
        alias: "helm-release2"
        typeRef: { path: "cap.type.helm.chart.release", revision: "0.1.0" }
        value: {
          key: "test" # same as it was
        }
        backend: {
          id: "123" # it's like that already.
          context: { # new property of type `Any!`
            name: "release-name",
            namespace: "release-namespace",
          }
        }
      }    
    ]
    usesRelations: []
  }
  ) {
    id
    alias
  }
}
```

The error is:

```
{
  "errors": [
    {
      "message": "Validation failed: Built-in storage backend does not accept context",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "createTypeInstances"
      ],
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR",
        "exception": {
          "stacktrace": [
            "Error: Validation failed: Built-in storage backend does not accept context",
            "    at createTypeInstances (/Users/pkosiec/repositories/capact/hub-js/dist/local/mutation/create-type-instances.js:24:15)",
            "    at /Users/pkosiec/repositories/capact/hub-js/node_modules/neo4j-graphql-js/node_modules/graphql-tools/node_modules/@graphql-tools/schema/index.cjs.js:689:24",
            "    at field.resolve (/Users/pkosiec/repositories/capact/hub-js/node_modules/apollo-server-core/dist/utils/schemaInstrumentation.js:56:26)",
            "    at resolveField (/Users/pkosiec/repositories/capact/hub-js/node_modules/graphql/execution/execute.js:464:18)",
            "    at /Users/pkosiec/repositories/capact/hub-js/node_modules/graphql/execution/execute.js:261:18",
            "    at /Users/pkosiec/repositories/capact/hub-js/node_modules/graphql/jsutils/promiseReduce.js:23:10",
            "    at Array.reduce (<anonymous>)",
            "    at promiseReduce (/Users/pkosiec/repositories/capact/hub-js/node_modules/graphql/jsutils/promiseReduce.js:20:17)",
            "    at executeFieldsSerially (/Users/pkosiec/repositories/capact/hub-js/node_modules/graphql/execution/execute.js:258:37)",
            "    at executeOperation (/Users/pkosiec/repositories/capact/hub-js/node_modules/graphql/execution/execute.js:236:55)"
          ]
        }
      }
    }
  ],
  "data": null
}
```

Reply:
I can add backend ID 👍  but alias is problematic as it's not required on input from user and in this case is automatically generated, so returning this alias won't help